SDLMTCloud.Provider/Sdl.Community.MTCloud.Provider/Service/MetadataSupervisor.cs

```diff
@@ -222,10 +222,11 @@ private void TranslationService_TranslationReceived(TranslationData translationD
 			}
 			else if (ActiveDocument is not null)
 			{
-				foreach (var sourceSegment in translationData.SourceSegments)
+				foreach (var sourceSegment in translationData.Segments)
 				{
 					var currentSegmentPair =
```

Comment:
do we check later for for null currentSegmentPair?

Reply:
Yes we do - directly in the method that uses it - AddTargetSegmentMetaData